### PR TITLE
Adds grant claim process to Rewards Panel

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/1783

screen shot 2018-12-13 at 2 11 08 pm
screen shot 2018-12-13 at 2 11 21 pm
screen shot 2018-12-13 at 2 11 32 pm

(See https://github.com/brave/brave-ui/pull/306#issue-236094082 for comments regarding design)

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
1. Build Brave from this branch and run with a fresh profile
2. Navigate to a website such as `brave.com`
3. Click the `claim` button in the grant notification
4. Test to ensure the captcha fails appropriately
5. Test that the captcha can be dismissed and re-opened
6. Test to ensure the captcha can be solved appropriately
7. Test to ensure that captcha success notification can be dismissed
8. Ensure funds are added to balance as expected.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/resources/extension/brave_rewards/background/reducers/grant_panel_reducer.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+              ...state,
+              grant
+            }
+          }
+        }
+        break
+      }
+    }
+
+  if (state !== startingState) {
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/background/reducers/grant_panel_reducer.ts]: This is not needed as we have save in index.ts. Please remove it from `rewards_panel_reducer` as well

Comment (ryanml) [reply components/brave_rewards/resources/extension/brave_rewards/background/reducers/grant_panel_reducer.ts]: done

**Comment by ryanml** [comment]
@NejcZdovc fix made in brave-ui, ready for another review

**Comment by ryanml** [comment]
(0.61x): 0f40d174d25895ebd13182cf49efa87f66b3f8ee

**Pull request merged**